src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -383,11 +383,21 @@ public override Task<byte[]> GetArchiveHashAsync(HashAlgorithmName hashAlgorithm
 
         public override bool CanVerifySignedPackages(SignedPackageVerifierSettings verifierSettings)
         {
+            var value = false;
 #if IS_DESKTOP
-            return true;
-#else
-            return false;
+            if (RuntimeEnvironmentHelper.IsMono) {
+                if (RuntimeEnvironmentHelper.IsWindows) {
+                    value = true;
+                }
+                else {
+                    value = false;
+                }
+            }
+            else {
+                value = true;
+            }
 #endif
+            return value;
```

Comment:
The tests that exist could potentially change, they are not asserting this scenario. 

Can you please add a test that specifically tests this scenario.

Reply:
@nkolev92 There's a new test specific for CanVerifySignedPackages;  testing both Destkop/Non-Desktop scenarios.